tests/test_datasets.py

```diff
@@ -340,3 +340,55 @@ def test_Ptycho2DDataset_downsample(test_ptycho_cxis):
                 assert np.allclose(np.array(dataset.background.shape) // factor,
                                    np.array(copied_dataset.background.shape))
         
+
+def test_Ptycho2DDataset_crop_2D_translations(ptycho_cxi_1):
+    # Grab dataset
+    cxi, expected = ptycho_cxi_1
+    dataset = Ptycho2DDataset.from_cxi(cxi)
+    copied_dataset = deepcopy(dataset)
+
+    # Test 1: Complain when the the bounds of an ROI are correctly defined,
+    #   but it does not contain any sample positions inside of it. The 
+    #   translations in ptycho_cxi_1 ranges from 0m to -300m in both x and y 
+    #   (it looks like a line scan). We select an ROI at (0, -300) which should 
+    #   not contain any translation positions.
+    with pytest.raises(ValueError) as excinfo:
+        copied_dataset.crop_2D_translations(roi=(0,-5,-295,-300))
+    assert('(i.e., patterns and translations will be empty)') in str(excinfo.value)
+
+    # Test 2: Draw an ROI that's centered in the middle of the x/y translation range,
+    #   but complain because the bounds of the ROI are incorrectly defined
+    x_left, y_top = dataset.translations[10,:2]
+    x_right, y_bottom = dataset.translations[-11,:2]
+
+    with pytest.raises(ValueError) as excinfo:
```

Comment:
See earlier comment - I think it just makes more sense to add a check so this works, instead of raising an error.

Reply:
I've reworked the test to check if the same result is obtained if the order in which the left and right (and/or bottom and top) elements in 'roi' are swapped.